src/lib/Runner.ts

```diff
@@ -269,10 +269,11 @@ export class Runner {
             sourceBranch: pullRequest.sourceBranch,
             targetBranch: pullRequest.targetBranch,
           });
-          await landRequest.setStatus(
-            'fail',
-            `Unable to merge pull request ${result.reason ? `: ${result.reason}` : ''}`,
-          );
+          const error =
+            result.reason?.error?.message +
+              '. ' +
+              result.reason?.error?.fields?.merge_checks?.join(', ') ?? '';
+          await landRequest.setStatus('fail', 'Unable to merge pull request: ' + error);
```

Comment:
Nit: This is easier to read. String literal is much nicer to maintain cause you can read it as a real sentence instead of checking code back and forth
```suggestion
          await landRequest.setStatus(
            'fail',
            `Unable to merge pull request: ${result.reason?.error?.message}. ${result.reason?.error?.fields?.merge_checks?.join(', ') ?? ''}`,
          );
```

Reply:
Also because it's `result.reason?.error?.message` what if anything along the optional chaining is `undefined`? We'd have a malformed error messaging

Reply:
It would print nothing instead since we're using the nullish coalescing operator